### PR TITLE
WIP shorten auto-generated table names

### PR DESCRIPTION
**Pull request by flamerohr**
Fix add check for legacy long table name and rename to short version

Fixes https://github.com/silverstripe/silverstripe-framework/issues/6904 and warning added for 4.0.1 https://github.com/silverstripe/silverstripe-framework/pull/7620

### CODE REVIEW

File: src/ORM/DataObject.php [hunk 1]
```diff
@@ -3064,13 +3064,31 @@ public function requireTable()
...
         }
 
+        if ($legacyTables) {
+            $ignore = Config::inst()->get(static::class, 'ignored_legacy_tables') ?: [];
+            $renameTables = array_diff(
+                array_intersect($legacyTables, DB::table_list()),
+                $ignore
+            );
+            if (count($renameTables) > 1) {
+                throw new LogicException(
```
Comment (tractorcow) [inline src/ORM/DataObject.php]: Maybe just make this a warning and keep doing the update, otherwise users may end up with an inconsistent database.

Comment (flamerohr) [reply src/ORM/DataObject.php]: hmm, I guess the issue I have is if there is more than one legacy table - which one would be the better one to use?

Comment (tractorcow) [reply src/ORM/DataObject.php]: Don't do any rename; Create a brand new table in that case? It's equally likely that none of the tables are valid as is one of them is.

**Comment by tractorcow** [comment]
@flamerohr I've done a bit of work on master to fix up the build. Rebasing in hope that this addresses the broken build. :)

**Comment by tractorcow** [comment]
Nope, I messed up the constraints... needs to be 2.x-dev not 2.0.x-dev /sigh

**Comment by tractorcow** [comment]
Ready to merge on green.

**Pull request merged**